src/embind/embind_gen.js

```diff
@@ -377,11 +382,12 @@ var LibraryEmbind = {
 
     print() {
       const out = ['{\n'];
+      const emitted = {};
```

Comment:
How about `emitted = new Set()` and then you can use `emitted.add(signature)` above ?